docs/rules/array-bracket-newline.md

```diff
@@ -0,0 +1,243 @@
+# enforce line breaks after open and before close array brackets (array-bracket-newline)
+
+A number of style guides require or disallow line breaks inside of array brackets.
+
+## Rule Details
+
+This rule enforces line breaks after open and before close array brackets
+
+## Options
+
+This rule has either a string option:
+
+* `"always"` requires line breaks inside braces
+* `"never"` disallows line breaks inside braces
+
+Or an object option:
+
+* `"multiline": true` (default) requires line breaks if there are line breaks inside properties or between properties
+* `"minItems"` requires line breaks if the number of properties is more than the given integer
+
+### always
+
+Examples of **incorrect** code for this rule with the `"always"` option:
+
+```js
+/*eslint array-bracket-newline: ["error", "always"]*/
+/*eslint-env es6*/
+
+let a = [];
+let b = [1];
+let c = [1, 2];
+let d = [1,
+    2];
+let e = [function foo() {
+    dosomething();
+}];
+```
+
+Examples of **correct** code for this rule with the `"always"` option:
+
+```js
+/*eslint array-bracket-newline: ["error", "always"]*/
+/*eslint-env es6*/
```

Comment:
I think @not-an-aardvark means that we could use `var`s instead here, and get rid of the `eslint-env` comment, just to keep the examples more succinct and straight to the point.